mypy/stubgenc.py

```diff
@@ -307,29 +302,38 @@ def is_skipped_attribute(attr: str) -> bool:
                     '__weakref__')  # For pickling
 
 
-def infer_method_sig(name: str) -> str:
+def infer_method_sig(name: str) -> List[TypedArgSig]:
     if name.startswith('__') and name.endswith('__'):
         name = name[2:-2]
         if name in ('hash', 'iter', 'next', 'sizeof', 'copy', 'deepcopy', 'reduce', 'getinitargs',
                     'int', 'float', 'trunc', 'complex', 'bool'):
-            return '()'
+            return []
         if name == 'getitem':
-            return '(index)'
+            return [TypedArgSig(name='index', type=None, default=False)]
```

Comment:
If you would use normal classes instead of named tuples you could define `__init__` as:
```python
    def __init__(self, name: str, type: Optional[str] = None, default: bool = False) -> None: ...
```
and then this an d a dozen others will be just `TypedArgSig('index')`.

Reply:
Also with normal classes you can define `__repr__()`, to simplify test cases significantly. You can just check that `str(<generated signature>)` matches an expected string.